src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,304 @@
+    },
+    model: {
+        contextId: "",
+        contextToCopy: "",
+        contextNames: [],
+        fieldErrors: ""
+    },
+    selectors: {
+        contextIdInput: "#pmt-add-context-name-input",
+        contextToCopySelect: "#pmt-context-to-copy-select"
+    },
+    bindings: {
+        contextIdInput: "contextId",
+        contextToCopySelect: "contextToCopy"
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.addContextDialog.acceptConfirmDialog",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"]
+        }
+    }
+});
+
+/**
+ * Dialog for adding a new preference set (context) to a preferences safe (prefset).
+ * Includes a dropdown that can be used to select a context to copy the
+ * initial settings from.
+ *
+ * @param  {that} that           Add Context Dialog instance
+ * @param  {editPrefs} editPrefs Central gpii.devpmt.editPrefs component for the page.
+ */
+gpii.devpmt.dialogs.addContextDialog.acceptConfirmDialog = function (that, editPrefs, event) {
+    // We don't want the form to actually submit the page, just to enable
+    // activating the submit button on Enter
+    event.preventDefault();
+
+    // Validate context name, eventually this will be replaced with
+    // our new json schema work.
+    if (that.model.contextId === "") {
+        that.applier.change("fieldErrors", "Please enter a name.");
+        that.reRender();
+        return;
+    }
+    else if (!/^\w+$/.exec(that.model.contextId)) {
+        that.applier.change("fieldErrors", "Please use only alphanumeric characters.");
+        that.reRender();
+        return;
+    }
+    else {
+        // In case we are revalidating on a subsequent attempt
+        that.applier.change("fieldErrors", "");
+    }
+
+    // TODO validation to see if already exists, and determining
+    // the valid set of strings a context ID can take
+    var contextToCopy = that.model.contextToCopy;
+    var path = "flatPrefs.contexts." + that.model.contextId;
+    var newPrefSet = {};
+    // The select contains a blank option to start with a fresh set of values.
+    if (that.model.contextToCopy !== "" && editPrefs.model.flatPrefs.contexts[contextToCopy]) {
+        newPrefSet = fluid.copy(editPrefs.model.flatPrefs.contexts[contextToCopy].preferences);
+    }
+    editPrefs.applier.change(path, {
+        "name": that.model.contextId,
+        "preferences": newPrefSet
+    }, "ADD");
+    that.closeDialog();
+};
+
+/**
+ * Edit NP Set Context Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.editContextDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-editContext-dialog"
+    },
+    model: {
+        // Store so that we can fetch the context to copy to a new ID
+        originalContextId: "",
+        contextId: "",
+        contextIdErrors: "",
+        contextName: "",
+        contextNameErrors: "",
+        fieldErrors: ""
```

Comment:
`fieldErrors` is not used in this component. Can it be removed from here?

Reply:
removed